core/src/main/java/org/wso2/testgrid/tinkerer/UnixClient.java

```diff
@@ -0,0 +1,157 @@
+                                        StringUtil.concatStrings(this.getTinkererUserName(),
+                                                ":", this.getTinkererPassword()).getBytes(Charset.defaultCharset())))
+                        .bodyString("{\"code\":\"SHELL\",\"request\":\"ls " + logLocation + "\"}",
+                                ContentType.APPLICATION_JSON).execute();
+                Map<String, String> resultMap = new Gson().fromJson(logFiles.returnContent().asString(),
+                        new PropertyType().getType());
+                if (resultMap.containsKey("response")) {
+                    String response = resultMap.get("response");
+                    List<String> logFileNames = Arrays.asList(response.split("\n"));
+
+                    for (String logFileName : logFileNames) {
+                        logger.info("Downloading log file : " + logFileName);
+                        String key = Base64.getEncoder()
+                                .encodeToString(FileUtils.readFileToByteArray(new File(testPlan.getKeyFileLocation())));
+                        String source = logLocation + logFileName;
+                        String destination = TestGridUtil.getTestGridHomePath() + File.separator +
+                                testPlan.getDeploymentPattern().getProduct().getName() + logFileName;
+                        String bastianIP = deploymentCreationResult.getBastianIP();
+                        //TODO set correct download path so the log files are in the correct place in the folder hierarchy
+                        String downloadPath = this.getTinkererBase() + "test-plan/" + agent.getTestPlanId() +
+                                "/agent/" + agent.getInstanceName() + "/stream-file";
+                        Request.Post(downloadPath)
+                                .setHeader(HttpHeaders.CONTENT_TYPE, "application/json")
+                                .bodyString("{" +
+                                                "\"code\":\"STREAM_FILE\"," +
+                                                "\"data\":{\"key\":\"" + key + "\"," +
+                                                "\"source\":\"" + source + "\"," +
+                                                "\"destination\":\"" + destination + "\"" +
+                                                (bastianIP != null ? ",\"bastian-ip\":\"" +
+                                                        bastianIP + "\"" : "") + "}}"
+                                        , ContentType.APPLICATION_JSON)
+                                .execute();
+                        logger.debug("Download Location :" + destination);
+                    }
+                    logger.info("Successfully downloaded all log files ");
+                }
+            }
+        } catch (IOException e) {
+            throw new TinkererOperationException("Error occurred while executing the Tinkerer operation ", e);
```

Comment:
Shall we add some more details to the Error such as , operation name, error occurred in which agent from the deploymentCreationResult, log file name, etc.. :)